client-v2/src/strategies.ts

```diff
@@ -0,0 +1,38 @@
+import { frontsEdit } from 'constants/routes';
+import { matchPath } from 'react-router';
+import {
+  fetchEditionsIssueAsConfig,
+  fetchFrontsConfig
+} from 'services/faciaApi';
+
+const isValidPathForEdition = (priority: string, id?: string): id is string =>
+  !!id && priority === 'edition';
+
+const isValidPathForOtherPriority = (priority: string, id?: string) =>
+  !id && priority !== 'edition';
+
+const fetchFrontsConfigStrategy = (v2Path: string) => {
```

Comment:
Super minor: I wonder whether it'd be nicer to have the application state as a dependency here, rather than v2Path? We can then select out v2Path ourselves. Then, as a developer, all you'd have to understand is the `state` type, which the IDE could give you advice on, rather than have to reason about the path type, which is stringly typed and possible to get wrong.

Reply:
This is the right thing to do, it was a _slightly_ easier refactor as this was the dependency of the function for the first implementation of this - but I did wonder if I should do it as I was pushing this PR.